src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -570,5 +576,85 @@ public uint GetTokens(uint dwMaxTokens, IVsSearchToken[] rgpSearchTokens)
                 return 0;
             }
         }
+
+        private class PackageInstallerServicesProxy : IPackageInstallerServicesProxy
```

Comment:
Well, instead of creating proxy interfaces 1:1 with the existing NuGet interfaces, you could instead make a single interface that has the API you want, and have a single class that implements that, by wrapping the 4 interfaces from NuGet.

(Aka - the [adapter pattern](https://en.wikipedia.org/wiki/Adapter_pattern))